django/contrib/admin/options.py

```diff
@@ -1950,6 +1951,31 @@ def history_view(self, request, object_id, extra_context=None):
             "admin/object_history.html"
         ], context)
 
+    def get_changeformset_initial_data(self, request, prefix_name):
+        """
+        Get the initial formset data from the request's GET params.
+        """
+        initial = dict(request.GET.items())
+        initial_list = []
+        for k in initial:
+            args_list = k.split(sep='_')
+            if re.match(f'{prefix_name}__*_*', k):
+                try:
+                    f = self.model._meta.get_field(args_list[-1])
+                    try:
+                        initial_list[int(args_list[-2]) - 1][args_list[-1]] = initial[k]
+                        # We have to special-case M2Ms as a list of comma-separated PKs.
+                        if isinstance(f, models.ManyToManyField):
+                            initial_list[args_list[-2] - 1][args_list[-1]] = initial[k].split(',')
+                    except IndexError:
+                        initial_list.append({args_list[-1]: initial[k]})
+                        # We have to special-case M2Ms as a list of comma-separated PKs.
+                        if isinstance(f, models.ManyToManyField):
+                            initial_list.append({[args_list[-1]]: initial[k].split(',')})
+                except FieldDoesNotExist:
+                    continue
+        return initial_list
```

Comment:
I feel that if a functionality for passing formset initial parameters through GET request URL would be added, then it would be convenient for administrators or developers.
I feel that the implementation is rather very complex. May you suggest something to improve the implementation of the function and to reduce the complexity?

Reply:
@manav014 I guess you don't have the code anymore since you have force pushed the branch? This functionality is exactly what I need (And funnily enough, chatgpg claimed that this would work out of the box :D)

Reply:
<img width="697" alt="image" src="https://github.com/django/django/assets/41973254/d8d08dea-ba19-4b39-934f-41c5e08dee2e">

> Remember that this method relies on the current behavior of the Django admin interface

The ai is hallucinating :D

Reply:
Hey @autoantwort 
My branch is still not deleted from the fork you can refer to that for the commit history.

https://github.com/manav014/django/commits/ticket_26607/